articles/deploying-machine-learning-models-using-flask/index.md

```diff
@@ -0,0 +1,299 @@
+date: 2020-12-00T00:00:00-00:00
+topics: []
+excerpt_separator: <!--more-->
+images:
+  - url: /engineering-education/deploying-machine-learing-models-using-flask/hero.jpg
+    alt: Deploying Machine Learing models using Flask
+---
+In this article, we will learn about deploying Machine Learning models using Flask. By the end of the article, you will have an overview of how Machine Learning models are built, how Flask servers interact with our Machine Learning model, and how to connect the model with a web application. You will also learn a step-by-step procedure for deploying the model locally.
+<!--more-->
+I highly recommend going over [this](/engineering-education/complete-guide-on-installing-flask-for-beginners/) article to better grasp those concepts regarding Flask. As a prerequisite, a little knowledge about [HTML](https://en.wikipedia.org/wiki/HTML) and [CSS](https://en.wikipedia.org/wiki/CSS) would help to follow this article along.
+
+### Table of contents
+
+- [Objective](#objective)
+- [Machine Learning model](#machine-learning-model)
+- [HTML webpage](#html-webpage)
+- [Flask webserver](#flask-webserver)
+- [Output display page](#output-display-page)
+- [Conclusion](#conclusion)
+- [Further reading](#further-reading)
+
+### Step by step guide for implementation
+#### Objective
+In this tutorial, we are going to deploy a simple Machine Learning model using the flask webserver. For the demonstration, our Machine Learning model helps us classify the variety of flowers based on the length and width of sepals and petals. So, we build a simple HTML webpage to accept the measurements as input and classify the variety based on the classification model.
+
+#### Machine Learning model
+##### Dataset
+For building the Machine Learning model, we make use of the [Iris](https://gist.github.com/netj/8836201) dataset. A glimpse of the data is as shown below:
+
+![A few rows of Iris dataset](/engineering-education/deploying-machine-learing-models-using-flask/Dataset.PNG)
+
+##### Classification model
+Having chosen the dataset, it's time to build our classification model. Let's name the file `model.py`.
+
+Firstly, we import the necessary Python libraries for building the classification model. Here, we use the following libraries:
+
+- Pandas.
+
+- Numpy.
+
+- Sklearn/Sci-kit learn,
+
+```python
+# Importing necessary libraries
+import pandas as pd # To manage data as data frames
+import numpy as np # To manipulate data as arrays
+from sklearn.linear_model import LogisticRegression # Classification model
+```
+
+After importing the libraries, we import the dataset using `pd.read_csv` command. This command reads the CSV file and transforms it into a data frame called `data`. Generally, working with data frames is much easier when compared to that of arrays.
+
+The CSV file of the dataset can be downloaded [here](https://gist.github.com/netj/8836201).
+
+```python
+# Importing the dataset
+data = pd.read_csv('[YOUR-ABSOLUTE-PATH]/iris.csv')
```

Comment:
This might cause confusion to absolute beginners. Instead, use a relative path and ask them to store both the code and csv file in the same directory.

Reply:
Done